src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/HaplotypeCallerEngine.java

```diff
@@ -811,54 +717,8 @@ public void shutdown() {
-        if ( region.isFinalized() ) {
-            return;
-        }
-
-        // Loop through the reads hard clipping the adaptor and low quality tails
-        final List<GATKRead> readsToUse = new ArrayList<>(region.getReads().size());
-        for( final GATKRead myRead : region.getReads() ) {
-            GATKRead clippedRead;
-            if ( hcArgs.errorCorrectReads ) {
-                clippedRead = ReadClipper.hardClipLowQualEnds(myRead, MIN_TAIL_QUALITY_WITH_ERROR_CORRECTION);
-            }
-            else { // default case: clip low qual ends of reads
-                clippedRead = ReadClipper.hardClipLowQualEnds(myRead, minTailQuality);
-            }
-
-            if ( hcArgs.dontUseSoftClippedBases || ! ReadUtils.hasWellDefinedFragmentSize(clippedRead) ) {
-                // remove soft clips if we cannot reliably clip off adapter sequence or if the user doesn't want to use soft clips at all
-                clippedRead = ReadClipper.hardClipSoftClippedBases(clippedRead);
-            }
-            else {
-                // revert soft clips so that we see the alignment start and end assuming the soft clips are all matches
-                // TODO -- WARNING -- still possibility that unclipping the soft clips will introduce bases that aren't
-                // TODO -- truly in the extended region, as the unclipped bases might actually include a deletion
-                // TODO -- w.r.t. the reference.  What really needs to happen is that kmers that occur before the
-                // TODO -- reference haplotype start must be removed
-                clippedRead = ReadClipper.revertSoftClippedBases(clippedRead);
-            }
-
-            clippedRead = clippedRead.isUnmapped() ? clippedRead : ReadClipper.hardClipAdaptorSequence(clippedRead);
-            if ( ! clippedRead.isEmpty() && clippedRead.getCigar().getReadLength() > 0 ) {
-                clippedRead = ReadClipper.hardClipToRegion( clippedRead, region.getExtendedSpan().getStart(), region.getExtendedSpan().getEnd() );
-                if ( region.readOverlapsRegion(clippedRead) && clippedRead.getLength() > 0 ) {
-                    readsToUse.add(clippedRead);
-                }
-            }
-        }
-
-        // TODO -- Performance optimization: we partition the reads by sample 4 times right now; let's unify that code.
-        // final List<GATKRead> downsampledReads = DownsamplingUtils.levelCoverageByPosition(ReadUtils.sortReadsByCoordinate(readsToUse), maxReadsInRegionPerSample, minReadsPerAlignmentStart);
-        Collections.sort(readsToUse, new ReadCoordinateComparator(readsHeader)); // TODO: sort may be unnecessary here
-
-        // handle overlapping read pairs from the same fragment
-        cleanOverlappingReadPairs(readsToUse);
-
-        region.clearReads();
-        region.addAll(readsToUse);
-        region.setFinalized(true);
+    private void finalizeRegion(final AssemblyRegion region) {
```

Comment:
Now that `finalizeRegion` is in the util class, we should get rid of this private method and prepend the util class (`AssemblyBasedCallerUtils.finalizeRegion()`) when we call it.